.github/workflows/tests.yml

```diff
@@ -24,17 +24,16 @@ jobs:
 
       # python setup
       - name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v1
+        uses: actions/setup-python@v4
         with:
           python-version: ${{ matrix.python-version }}
+          cache: 'pip'
 
-      # python cache
-      - uses: actions/cache@v1
+      # buildout eggs cache
+      - uses: actions/cache@v3
         with:
-          path: ~/.cache/pip
-          key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements.txt') }}
-          restore-keys: |
-            ${{ runner.os }}-pip-
+          path: eggs
+          key: ${{ runner.os }}-py${{ matrix.python-version }}-plone${{ matrix.plone-version }}-buildout-
```

Comment:
This is close to what we use in [coredev](https://github.com/plone/buildout.coredev/blob/6.0/.github/workflows/run-buildout.yml). I would add some restore-keys though. If I have it correctly in my head, if the key matches exactly then it is restored. Otherwise GitHub looks for the closest matching restore key. Here I guess only one makes sense:

```
restore-keys: |
    ${{ runner.os }}-py${{ matrix.python-version }}-
```

This should mean that when GitHub actions runs the 6.0 tests and it does not find the exact key, say `linux-3.8-plone6.0-buildout-` then it looks for keys starting with `linux-3.8-` and it might find `linux-3.8-plone5.2-buildout-`. Then you get an eggs directory with eggs for 5.2. At a guess, about one fourth of these can be reused on 6.0, so running buildout takes roughly one fourth less time.

When I think about it, I expect that when all goes well in such a run, at the end a new cache gets stored and this then contains all 6.0 eggs from the new buildout run, plus the previous 5.2 eggs. So the cache is bigger than it has to be. This may or may not be what you want. I would say it is okay.

Reply:
@mauritsvanrees Thanks for the suggestion. I thought about this, but I figured it's not a big deal if there is a cache miss and everything is downloaded again from PyPI. After all, that's what we've been doing on every build so far.

I'm more worried about what happens when there is a minor Plone version bump, which does not currently change the cache key. Then there will be a cache hit, but with older versions. That's okay, because buildout will download the new ones, but because there was a cache hit, the cache action will not update the cache. So it will be slower again on every build.

To avoid that I guess we would need to put the complete Plone version somewhere it can be easily extracted to use in the cache key. Probably not that hard but I didn't spend time trying to solve it yet.

Reply:
Good point. In that case adding `${{ hashFiles('versions*.cfg') }}` to the key should help. That hash will change for every Plone version because we explicitly specify the Plone version in it again. So:

```
key: ${{ runner.os }}-py${{ matrix.python-version }}-plone${{ matrix.plone-version }}-buildout-${{ hashFiles('versions*.cfg') }}-
restore-keys: |
    ${{ runner.os }}-py${{ matrix.python-version }}-plone${{ matrix.plone-version }}-buildout-
    ${{ runner.os }}-py${{ matrix.python-version }}-
```

Reply:
@mauritsvanrees yeah, that's the right idea, but in this repo versions.cfg does not include the Plone versions. That's done in plone-6.0.x.cfg or plone-5.2.x.cfg. So it depends which buildout configuration is actually run...

Reply:
Sorry, I meant to say `hashFiles('*.cfg')`, or maybe `hashFiles('plone*.cfg')` could also work. This does mean that a change in `plone-5.2.x.cfg` causes a needless exact cache miss in the Plone 6.0 job. But the first `restore-key` will still match.

Reply:
@mauritsvanrees That sounds like it might work; I'll give it a try.